api/server/router/local/image.go

```diff
@@ -137,6 +137,12 @@ func (s *router) postImagesCreate(ctx context.Context, w http.ResponseWriter, r
 				err = s.daemon.PullImage(ref, metaHeaders, authConfig, output)
 			}
 		}
+		// Check the error from pulling an image to make sure the request
+		// was authorized. Modify the status if the request was
+		// unauthorized to respond with 401 rather than 500.
+		if err != nil && strings.Contains(err.Error(), "unauthorized") {
```

Comment:
Does it work to check for `errcode.ErrorCodeUnauthorized` (from `github.com/docker/distribution/registry/api/errcode`) and `registry.errLoginRequired` (needs to be exported) instead of looking in the text?

If we don't want to import vendored distribution code here, `Pull` in `github.com/docker/docker/distribution` could convert these to a single exported error value.